CMakeLists.txt

```diff
@@ -146,6 +146,29 @@ set(ALLEGRO_SOURCES
 )
 
 
+# Defining helper-function to deal with setups that manually set an
+# absolute path for CMAKE_INSTALL_(LIB|INCLUDE)DIR...
+# We could also just make all paths absolute, but this way the
+# pkg-config-file is more human-readable and pkg-config may be able to deal
+# with varying prefixes.
+function(make_pkg_config_absolute out_path in_path)
```

Comment:
Instead of this helper, you could use directly `CMAKE_INSTALL_FULL_LIBDIR` and `CMAKE_INSTALL_FULL_INCLUDEDIR` which are provided by GNUInstallDirs and are absolute versions of the non-full ones.

Reply:
Hmm... I could swear there was some reason for this, when I wrote this.... Let me double-check

Reply:
I see, that the "make_absolute" actually happens at runtime inside pkg-config by using $prefix instead of writing the absolute path to the file. The only effect I can see is that the pkg-config file wiil look cleaner, but I severely doubt that this was motivating me to write this ugly-Cmake-code.

Let me think about this some more time. If I cannot remember the reason, then it must indeed be me over-complicating things for no reason whatsoever.

Reply:
It's true that your logic is cleaner, as a "good" pkgconfig file should indeed have `${prefix}/lib` or `${prefix}/lib64` instead of a hardcoded prefix. But I've checked various pkgconfig files from CMake-based projects and it seems everybody goes the route of hardcoding the absolute path, which in the end doesn't really hurt either.

Reply:
Not only CMake-based projects btw, autotools ones too:
```
$ cat /usr/lib64/pkgconfig/flac.pc 
prefix=/usr
exec_prefix=/usr
libdir=/usr/lib64
includedir=/usr/include

Name: FLAC
Description: Free Lossless Audio Codec Library
Version: 1.3.2
Requires.private: ogg
Libs: -L${libdir} -lFLAC
Libs.private: -lm
Cflags: -I${includedir}
```

Reply:
Reading that comment above the code seems to hint towards relocatability of the pkg-config file. Under Windows pkg-config will automatically define the prefix to be two levels above the `.pc`-file. I think specifying an absolute path  for libdir may break this, because it no longer refers to prefix.

Does that make sense?

Reply:
That definitely makes sense. Windows is a bit uncharted territory for me as far as pkg-config goes :D

Reply:
Anyways, if even I (as the author of this code) cannot remember this after a year or so, the comment or the function-name needs to be changed :-)

Reply:
**TL;DR: I will remove this code, because pkg-config handles this automatically, nowadays**
This changed over pkg-config versions...
[Now (from 0.29 it seems) pkg-config defaults to "usually" (*What does that mean?*) overriding the prefix on all OSes](). It even intelligently will replace expanded variables, so we might indeed be able to remove this code all together. And it turns out this intelligent replacing took place even in older versions on Windows. This is news to me, so I am pretty sure that I was not aware of magic happening with the variables from the pkg-config-side that made my code here unnecessary. It seems the pkg-config-developers kind of adapted to developers hard-coding absolute paths here. Thanks for bringing this up.

[Previous versions (0.28) said](https://manpages.debian.org/jessie/pkg-config/pkg-config.1.en.html):

>    --dont-define-prefix
        This option is available only on Windows. It prevents pkg-config from automatically trying to override the value of the variable "prefix" in each .pc file. 
[...]
**WINDOWS SPECIALITIES**
> If a .pc file is found in a directory that matches the usual conventions (i.e., ends with \lib\pkgconfig or \share\pkgconfig), the prefix for that package is assumed to be the grandparent of the directory where the file was found, and the prefix variable is overridden for that file accordingly.
>
>If the value of a variable in a .pc file begins with the original, non-overridden, value of the prefix variable, then the overridden value of prefix is used instead. 

Some older versions  (I downloaded a random old tarball to check) do not have the last sentence about this automatic replacement in their manpage.

Reply:
fixed in new commit